src/utils/utils.js

```diff
@@ -188,3 +188,65 @@ export function getUrlQuery() {
         window.location.search || window.location.hash.replace('#/map?', '')
     )
 }
+
+/**
+ * Parse a URL hash fragment
+ *
+ * We use query in the hash fragment, the standard new URL doesn't parse the query from the hash,
+ * this function does and return the URL object with the hash fragment without query and the the
+ * query separated.
+ *
+ * @param {string} url
+ * @returns {{ urlObj: URL; hash: string; query: string }} Parsed url
+ */
+export function parseUrlHashQuery(url) {
+    const urlObj = new URL(url)
```

Comment:
this might raise a `TypeError` if the given URL isn't valid, I don't know if it wouldn't be better to wrap all this in a `try...catch` and manage the error here (instead of giving it back to the caller)

Or we decide this is a purely internal function, we know what we are doing, and it is while developing that we have error in the console and all is "well"

Reply:
I can add a try catch, might be a good idea for future. Currently the function is only called with window.location.href which is by design always valid

Reply:
After looking close I think it is better to leave it as is. I think it would be wise to do the try catch if the input comes from a user (like for the external provider), but here it is for the moment used to programmatisch change the url from map to embed or vise versa and the url should always be valid. If it is not, then it is a code mistake and it is better to crash early with a backtrace as silently in my opinion.